presto-hive/src/main/java/com/facebook/presto/hive/util/ConfigurationUtils.java

```diff
@@ -73,8 +77,19 @@ public static JobConf toJobConf(Configuration conf)
 
     public static JobConf configureCompression(Configuration config, HiveCompressionCodec compression)
     {
-        JobConf result = new JobConf(false);
-        copy(config, result);
+        JobConf result;
+        // Workaround for https://github.com/prestodb/presto-hadoop-apache2/commit/aa3a59101cb14659a96400a3943cd1bb740399b9
+        // This code assumes that if configuration is an instance of FileSystemFactory, a new instance is always supplied
+        if (config instanceof FileSystemFactory) {
+            checkArgument(config instanceof JobConf, "config is not an instance of JobConf: %s", config.getClass());
+            checkArgument(config.get(IS_NOT_A_COPY) == null, "config is not a copy");
```

Comment:
i don't understand this. What's the is_not_a_copy flag, and what does the error message here mean?

Reply:
also, does this commit need to get squashed into add compression_codec session property for correctness purposes?

Reply:
> What's the is_not_a_copy flag, and what does the error message here mean?

Basically that's a sanity check to verify that the configuration passed to this method is always a clean copy.

Reply:
> also, does this commit need to get squashed into add compression_codec session property for correctness purposes?

Sure, i don't mind squashing it once the review is done. Just wanted to extract it to a separate commit so i can provide a more maningful explanation in the commit message.

Reply:
I'm not sure I understand. Here's what I think you're saying:
"is_not_a_copy" is  your own made up flag to indicate if we are dealing with a new instance of a jobconf (a copy) or whether we have a shared instance. 
1. if "is_not_a_copy" is true, that means we have a shared instance
2. if it's false then it's a new unique instance
3.  and if it's null/unset that also means it's unique because we assume we always get a new instance from FileSystemFactory?  (if this is true, the error message for the checkArgument needs to be fixed).

I still don't understand why we care.  If mutating a shared JobConf is a problem, why don't we create our own copy in the method? that seems much more standard.  Also, why do we set it to true next? all that we do with it  is set some properties in a helper method.

Reply:
Discussed offline. Decided to remove the assertion check, as it looks incredibly confusing.